### PR TITLE
Add templates for Github

### PR DESCRIPTION
**Pull request by aerorahul**
This PR:
- adds templates for creating bug reports, new issues and pull requests.
- adds `*.sw[a-p]` to ignore `vi` swap files in `.gitignore`

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/pull_request.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+-->
+
+# Checklist:
+
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
```
Comment (WalterKolczynski-NOAA) [inline .github/ISSUE_TEMPLATE/pull_request.md]: Given we don't have formal tests set up yet, might want to remove these two for now.

File: .github/ISSUE_TEMPLATE/pull_request.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+name: Pull request
+about: Use this template to open pull requests
+title: '<title>'
+labels:
+assignees: 'KateFriedman-NOAA', 'WalterKolczynski-NOAA'
+
+---
+
```
Comment (WalterKolczynski-NOAA) [inline .github/ISSUE_TEMPLATE/pull_request.md:9]: Before the description section, add directions along these lines:

Before opening a PR, please note these guidelines:
- Each PR should only address ONE topic and have an associated issue
- No hardcoded or paths to personal directories should be present
- No temporary or backup files should be committed
- Any code that was disabled by being commented out should be removed

File: .gitignore [hunk 3]
```diff
@@ -4,6 +4,7 @@ __pycache__
 *.pyc
 *.[aox]
 *.mod
+*.sw[a-p]
```
Comment (WalterKolczynski-NOAA) [inline .gitignore:7]: Does emacs have similar temporary files we should ignore?

**Review by WalterKolczynski-NOAA — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/pull_request.md (hunk 1 above)
Comment (aerorahul) [reply .github/ISSUE_TEMPLATE/pull_request.md]: Sounds good.

File: .gitignore (hunk 3 above)
Comment (aerorahul) [reply .gitignore:7]: I don't know about `emacs`.  I know some IDE's create `filename~`

File: .github/ISSUE_TEMPLATE/pull_request.md (hunk 2 above)
Comment (aerorahul) [reply .github/ISSUE_TEMPLATE/pull_request.md:9]: Will do.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Review by KateFriedman-NOAA — APPROVED** [review]
Looks great! I'm looking forward to using these. Thanks @aerorahul !

**Pull request merged**